### PR TITLE
#3429 [Bug Report] Transaction blocking error when creating payment f…

### PR DESCRIPTION
**Pull request by e-Evolution**
…rom the account statement and immediate posting  https://github.com/adempiere/adempiere/issues/3429

### CODE REVIEW

File: base/src/org/compiere/process/BankStatementPayment.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ private MPayment createPayment (int invoiceId, int bPartnerId,
 		payment.setDescription(description);
 		//
 		if (invoiceId != 0) {
-			MInvoice invoice = new MInvoice (getCtx(), invoiceId, null);
+			MInvoice invoice = new MInvoice (getCtx(), invoiceId, get_TrxName());
```
Comment (yamelsenih) [inline base/src/org/compiere/process/BankStatementPayment.java:231]: I just saw a change here but all class is changed, please review your head based on last commit of develop

File: base/src/org/compiere/process/BankStatementPayment.java [hunk 2]
```diff
@@ -151,21 +151,21 @@ private String createPayment (MBankStatementLine bankStatementLine) throws Excep
...
 			bankStatement.getC_BankAccount_ID(), bankStatementLine.getStatementLineDate(), bankStatementLine.getDateAcct(),
-			bankStatementLine.getDescription(), bankStatementLine.getAD_Org_ID(), bankStatementLine.getC_Charge_ID(), isUnidentified, documentNo, checkNo);
+			bankStatementLine.getDescription(), bankStatementLine.getAD_Org_ID(), bankStatementLine.getC_Charge_ID(), isUnidentified, documentNo, checkNo);
 		//	update statement
 		bankStatementLine.setPayment(payment);
-		if(isUnidentified) {
-			bankStatementLine.setC_Charge_ID(-1);
-		}
+		if(isUnidentified) {
+			bankStatementLine.setC_Charge_ID(-1);
```
Comment (yamelsenih) [inline base/src/org/compiere/process/BankStatementPayment.java:167]: This not apply

File: base/src/org/compiere/process/BankStatementPayment.java [hunk 3]
```diff
@@ -151,21 +151,21 @@ private String createPayment (MBankStatementLine bankStatementLine) throws Excep
...
+		}
 		//
 		MPayment payment = createPayment (bankStatementLine.getC_Invoice_ID(), bankStatementLine.getC_BPartner_ID(),
 			bankStatementLine.getC_Currency_ID(), bankStatementLine.getStmtAmt(), bankStatementLine.getTrxAmt(), 
 			bankStatement.getC_BankAccount_ID(), bankStatementLine.getStatementLineDate(), bankStatementLine.getDateAcct(),
-			bankStatementLine.getDescription(), bankStatementLine.getAD_Org_ID(), bankStatementLine.getC_Charge_ID(), isUnidentified, documentNo, checkNo);
+			bankStatementLine.getDescription(), bankStatementLine.getAD_Org_ID(), bankStatementLine.getC_Charge_ID(), isUnidentified, documentNo, checkNo);
 		//	update statement
 		bankStatementLine.setPayment(payment);
-		if(isUnidentified) {
```
Comment (yamelsenih) [inline base/src/org/compiere/process/BankStatementPayment.java:166]: This not apply

File: base/src/org/compiere/process/BankStatementPayment.java [hunk 4]
```diff
@@ -151,21 +151,21 @@ private String createPayment (MBankStatementLine bankStatementLine) throws Excep
 			documentNo = bankStatementLine.getEftReference();
 		}
 		String checkNo = bankStatementLine.getEftCheckNo();
-		boolean isUnidentified = false;
```
Comment (yamelsenih) [inline base/src/org/compiere/process/BankStatementPayment.java:154]: This not apply

File: base/src/org/compiere/process/BankStatementPayment.java [hunk 5]
```diff
@@ -269,9 +269,9 @@ private MPayment createPayment (int invoiceId, int bPartnerId,
 			payment.setCheckNo(checkNo);
 		}
 		payment.setIsUnidentifiedPayment(isUnidentified);
-		if(isUnidentified) {
-			payment.addDescription(Msg.parseTranslation(getCtx(), "@UnidentifiedPayment@"));
```
Comment (yamelsenih) [inline base/src/org/compiere/process/BankStatementPayment.java:273]: This not apply

**Review by yamelsenih — CHANGES_REQUESTED** [review]
Please change it

**Comment by e-Evolution** [comment]
@yamelsenih The last one that committed to this class was you, and it has a format for window, which is not the one agreed in the repository so in the future when you change a class make sure to send your commits in CRLF format

image


This causes problems in the format for the following developers who are respecting the standard of the agreed CRLF format

**Pull request closed without merging**